### DISCUSSION TITLE
Recommended Autoscaling Metric for ECS based Airflow setup

### DISCUSSION THREAD

**Discussion by chan64** [apache/airflow#24242] [Q&A]
I have setup Airflow environment in our AWS account using ECS, with webserver, scheduler, worker (along with celery) and flower as separate ECS services within the same cluster and Elasticache Redis as a broker.

For now, I am using CPU utilization as a metric for scaling the containers. I'm thinking "current number of messages in Message Broker" could be a good metric since number of tasks picked up by worker depends on it. Tried building custom autoscaling script but ran into issues with scaling down.

What is the recommended metric, especially for worker and scheduler? Why would it be so?

**Comment by potiuk** [comment]
See KEDA autoscaling in our Helm Chart https://airflow.apache.org/docs/helm-chart/stable/keda.html
